app/controllers/questions/consent_controller.rb

```diff
@@ -35,20 +35,20 @@ def after_update_success
 
       # the vita partner the client was routed to has capacity
       unless current_intake.client.routing_method_at_capacity?
-        tax_returns = []
-        TaxReturn.filing_years.each do |year|
-          tax_returns.push(TaxReturn.find_or_initialize_by(year: year, client: current_intake.client)) if current_intake.send("needs_help_#{year}") == "yes"
-        end
-        current_intake.client.tax_returns.replace(tax_returns)
-        tax_returns.map { |tr| tr.advance_to(:intake_in_progress) }
+        CreateInitialTaxReturnsService.new(intake: current_intake).create!
```

Comment:
nit: `InitialTaxReturnsService` would be fine too since you're calling create! on it.

Reply:
This is good because the service name is clearly too long.